### ISSUE TITLE
"not enough free GPU memory available" exception caught

### ISSUE THREAD

**Issue by lukkio88** [gpufit/Gpufit#47]
Hello, 

For some reason I don't quite understand I have an exception thrown due to the call of:

```
int gpufit
(
    size_t n_fits,
    size_t n_points,
    float * data,
    float * weights,
    int model_id,
    float * initial_parameters,
    float tolerance,
    int max_n_iterations,
    int * parameters_to_fit,
    int estimator_id,
    size_t user_info_size,
    char * user_info,
    float * output_parameters,
    int * output_states,
    float * output_chi_squares,
    int * output_n_iterations
)
try
{
    FitInterface fi(
        data,
        weights,
        n_fits,
        static_cast<int>(n_points),
        tolerance,
        max_n_iterations,
        static_cast<EstimatorID>(estimator_id),
        initial_parameters,
        parameters_to_fit,
        user_info, //0 in Gauss_Fit_2D example
        user_info_size, //0 in Gauss_Fit_2D example
        output_parameters,
        output_states,
        output_chi_squares,
        output_n_iterations);

    fi.fit(static_cast<ModelID>(model_id));

    return ReturnState::OK ;
}

catch( std::exception & exception )
{
    last_error = exception.what() ;

    return ReturnState::ERROR ;
}

```

Not entirely sure what the issue con be, I've implemented my own model to be minimized using an LSE estimator. What kind of information can I provide in order to get some help in debugging?

The content of `last_error` is `last_error = "not enough free GPU memory available"`

**Comment by adrianjp88** [comment]
Could you post an example code which producing this error?

**Comment by lukkio88** [comment]
Unfortunately I can't, but it is an extension of this example https://github.com/gpufit/Gpufit/issues/46. Can you just point out an effective way to debug it? I'm trying to use nvdia nsight but I'm having issues running it.

**Comment by lukkio88** [comment]
Anyway I partially solved by reducing the complexity of my model, where complexity in this context means actually the number of parameters of my model. I'll need a more complex one, but for purpose of testing for now it's fine. I assume the issue is actually number of parameters (simple as that I guess, given the vector/matrix of the gradient would be huge). Now I'm having other problems with the "calculate_hessian", it does return an error "invalid configuration argument". I'll post a separate issue.

**Comment by adrianjp88** [comment]
The exception "not enough free GPU memory available" is thrown before any call to a CUDA kernel function and GPU memory allocation.
I suppose, in your case the size of user info is huge. The available GPU memory for computation was reduced by the size of user info. I other words, until now, if the size of user info was too large, there was not enough memory available for computation. This is fixed by bf5d87a9675157ec233fd725c04df19f6bd60ca5.
